my-app/src/app/courses-page/directives/course-creation.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { Directive, ElementRef, Input, OnInit, Renderer2 } from '@angular/core';
+
+@Directive({
+  selector: '[appCourseCreation]'
+})
+export class CourseCreationDirective implements OnInit {
```

Comment:
since the pipes and directives are generic I would recommend to move them to `app/shared/pipes` and  `app/shared/directive` respectively. Also, don't hesitate to create separate modules for them)

Reply:
Done!